tcclient/creds.go

```diff
@@ -7,16 +7,75 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"net/http"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/taskcluster/jsonschema2go/text"
 	"github.com/taskcluster/slugid-go/slugid"
+	"github.com/tent/hawk-go"
 )
 
-type Credentials struct {
-	ClientId    string `json:"clientId"`
+type Credentials interface {
+	fmt.Stringer
+	ConfigureAuth(httpRequest *http.Request) error
+	getExtField() (string, error)
+	HawkCredentials() *hawk.Credentials
+}
+
+// Credentials for communicating with taskcluster services (Queue, Index,
+// Scheduler, etc).
+type TemporaryCredentials struct {
+	temporaryCredentials
+}
+
+type temporaryCredentials struct {
+	ClientID    string `json:"clientId"`
+	AccessToken string `json:"accessToken"`
+	// Certificate used only for temporary credentials
+	Certificate string `json:"certificate"`
+	// AuthorizedScopes if set to nil, is ignored. Otherwise, it should be a
+	// subset of the scopes that the ClientId already has, and restricts the
+	// Credentials to only having these scopes. This is useful when performing
+	// actions on behalf of a client which has more restricted scopes. Setting
+	// to nil is not the same as setting to an empty array. If AuthorizedScopes
+	// is set to an empty array rather than nil, this is equivalent to having
+	// no scopes at all.
+	// See http://docs.taskcluster.net/auth/authorized-scopes
+	AuthorizedScopes []string `json:"authorizedScopes"`
```

Comment:
ah, true...

--
Regards Jonas Finnemann Jensen.

2016-03-22 9:50 GMT-07:00 Pete Moore <notifications@github.com>:

> In tcclient/creds.go
> <https://github.com/taskcluster/taskcluster-client-go/pull/13#discussion_r57023323>
> :
>
> > +}
> > +
> > +type temporaryCredentials struct {
> > +	ClientID    string `json:"clientId"`
> > +	AccessToken string `json:"accessToken"`
> > +	// Certificate used only for temporary credentials
> > +	Certificate string `json:"certificate"`
> > +	// AuthorizedScopes if set to nil, is ignored. Otherwise, it should be a
> > +	// subset of the scopes that the ClientId already has, and restricts the
> > +	// Credentials to only having these scopes. This is useful when performing
> > +	// actions on behalf of a client which has more restricted scopes. Setting
> > +	// to nil is not the same as setting to an empty array. If AuthorizedScopes
> > +	// is set to an empty array rather than nil, this is equivalent to having
> > +	// no scopes at all.
> > +	// See http://docs.taskcluster.net/auth/authorized-scopes
> > +	AuthorizedScopes []string `json:"authorizedScopes"`
>
> Thanks Jonas. But if the type is not exported, the struct members are not
> visible outside the package, so there is no harm having them exported.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/taskcluster/taskcluster-client-go/pull/13/files/ecd79d67b954c7d0b21dc90c8a2ef95f6273a02e#r57023323>
>